src/main/java/core/basesyntax/service/FruitShopService.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.UnknownOperationException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitShopService {
+    private Map<String, Integer> fruitInventory = new HashMap<>();
+    private Map<OperationType, OperationHandler> operationStrategy = new HashMap<>();
+
+    public FruitShopService(Map<OperationType, OperationHandler> operationStrategy) {
+        this.operationStrategy.put(OperationType.BALANCE, new BalanceOperationHandler());
+        this.operationStrategy.put(OperationType.SUPPLY, new SupplyOperationHandler());
+        this.operationStrategy.put(OperationType.PURCHASE, new PurchaseOperationHandler());
+        this.operationStrategy.put(OperationType.RETURN, new ReturnOperationHandler());
```

Comment:
It appears that the constructor is overwriting the `operationStrategy` passed as a parameter with new instances of handlers. This defeats the purpose of dependency injection and does not follow the SOLID principles. Instead, you should use the passed `operationStrategy` map directly. This will allow for greater flexibility and testability of your `FruitShopService` class.